analyzers/src/SonarAnalyzer.CSharp/Rules/ConditionalSimplification.cs

```diff
@@ -283,38 +283,29 @@ private static bool AreTypesCompatible(ExpressionSyntax first, ExpressionSyntax
             return true;
         }
 
-        private static bool AreCandidateInvocations(ExpressionSyntax expression1,
-                                                    ExpressionSyntax expression2,
+        private static bool AreCandidateInvocations(ExpressionSyntax firstExpression,
+                                                    ExpressionSyntax secondExpression,
                                                     SyntaxNode comparedToNull,
                                                     SemanticModel semanticModel,
                                                     bool comparedIsNullInTrue)
         {
-            if (!(expression1 is InvocationExpressionSyntax methodCall1) || !(expression2 is InvocationExpressionSyntax methodCall2))
-            {
-                return false;
-            }
-
-            var methodSymbol1 = semanticModel.GetSymbolInfo(methodCall1).Symbol;
-            var methodSymbol2 = semanticModel.GetSymbolInfo(methodCall2).Symbol;
-
-            if (methodSymbol1 == null || methodSymbol2 == null || !methodSymbol1.Equals(methodSymbol2))
-            {
-                return false;
-            }
-
-            if (methodCall1.ArgumentList == null
-                || methodCall2.ArgumentList == null
-                || methodCall1.ArgumentList.Arguments.Count != methodCall2.ArgumentList.Arguments.Count)
+            if (firstExpression is not InvocationExpressionSyntax firstInvocation
+                || secondExpression is not InvocationExpressionSyntax secondInvocation
```

Comment:
I've simplified here the conditions a bit and merged them together. The code should should be faster too since now we get the symbols for the second invocation only if the first invocation is ok.